packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -113,10 +117,16 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
         } catch (error) {
             onUnhandledError(error);
         } finally {
-            this.setState({ isInitializing: false });
+            if (this.isShippingMounted) {
+                this.setState({ isInitializing: false });
```

Comment:
🍹 Instead of checking the mounted state, could we just cancel the XHR requests if the component is unmounted? e.g.:

```ts
import { createTimeout } from '@bigcommerce/checkout-sdk'; 

const timeout = createTimeout();

loadShippingOptions({ timeout });
```

And then in unmount function
```ts
timeout.complete();
```

And in the `catch` block, check if `error.status === 0` and not `setState` if the request is aborted.

Reply:
@davidchin When I tried to implement this, I found out that all three requests could complete before invoking `componentWillUnmount`, so checking the request error status might not work sometimes.

I ended up with another strategy to enhance the `setState` condition. That was to only `setState` if the shipping step was the active step.

Reply:
Testing video for the issue. Also tested the whole checkout process, working as expected.

https://user-images.githubusercontent.com/88361607/197689589-264c0609-b54c-4fd6-b85d-a7c2f2857d55.mov